### PR TITLE
Use Directory.Build.props/targets

### PR DESCRIPTION
**Pull request by natemcmaster**
Changes:

 - Removes the Import to build/common.props from all repos, relying instead on the automatic import of Directory.Build.props/targets
 - Move EFCore.Benchmarks.EFCore from test/ to benchmarks/. It's not really a test project.
 - Only uses Internal.AspNetCore.Sdk to src/ and test/. It's not necessary for the benchmarks projects.
 - Remove duplicate PackageReference's in test csproj files.

### CODE REVIEW

**Comment by natemcmaster** [comment]
Ping @bricelam @smitpatel can you take a look?

File: EFCore.Runtime.sln [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{CE6B50B2-34AE-44C9-940A-4E48C3E1B3BC}"
+	ProjectSection(SolutionItems) = preProject
+		src\Directory.Build.props = src\Directory.Build.props
+		src\EFCore.ruleset = src\EFCore.ruleset
+	EndProjectSection
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "test", "test", "{258D5057-81B9-40EC-A872-D21E27452749}"
+	ProjectSection(SolutionItems) = preProject
+		test\Directory.Build.props = test\Directory.Build.props
+		test\xunit.runner.json = test\xunit.runner.json
```
Comment (smitpatel) [inline EFCore.Runtime.sln]: drop xunit.runner.json

**Comment by natemcmaster** [comment]
> benchmarks on dev is still visible as project in EFCore.sln (& runtime edition). If not then can it be made?

Right now, two of the five benchmark projects are in EFCore.sln, none in EFCore.Runtime.sln. I didn't change which projects were in the sln files.

**Comment by natemcmaster** [comment]
PS - I'm out on vacation next week so I'll leave this one until I get back, or you can merge it yourself if you want. This one should go in before #9629

**Comment by smitpatel** [comment]
Ah. Right. Runtime does not contain benchmarks. Only Core solution.
:shipit:

**Comment by natemcmaster** [comment]
Rebased. I plan to merge this when CI is green.

File: build/common.props [hunk 2]
```diff
@@ -11,17 +12,15 @@
     <SignAssembly>True</SignAssembly>
     <VersionSuffix Condition="'$(VersionSuffix)' != '' And '$(BuildNumber)' != ''">$(VersionSuffix)-$(BuildNumber)</VersionSuffix>
     <DebugType Condition="'$(Configuration)' == 'Debug' AND '$(OS)' == 'Windows_NT'">full</DebugType>
-    <NoWarn>$(NoWarn);xUnit1004</NoWarn>
```
Comment (smitpatel) [inline build/common.props]: Leave this nowarn

File: build/common.props [hunk 3]
```diff
@@ -11,17 +12,15 @@
...
   </PropertyGroup>
+
   <ItemGroup>
     <EmbeddedResource Include="**\*.rd.xml" />
   </ItemGroup>
-  <ItemGroup>
-    <PackageReference Include="Internal.AspNetCore.Sdk" Version="$(InternalAspNetCoreSdkVersion)" PrivateAssets="All" />
-  </ItemGroup>
+
   <!-- TODO: Remove when TestDriven supports VS15.3+ -->
```
Comment (smitpatel) [inline build/common.props]: Could remove this now. we discussed in team meeting and everyone have moved to 15.3. Though optional. i will clean up if you don't want to.

File: build/common.props (hunk 2 above)
Comment (natemcmaster) [reply build/common.props]: It's in test/Directory.Build.props

File: build/common.props (hunk 3 above)
Comment (natemcmaster) [reply build/common.props]: I'll let you handle it this one :)

File: build/common.props (hunk 2 above)
Comment (smitpatel) [reply build/common.props]: Saw it just now. Thanks.

**Comment by smitpatel** [comment]
@natemcmaster - Can you open `OracleProvider.sln` and check if it builds. There are test projects in that solution which are not in EFCore.sln which are referencing common.props still.

**Comment by natemcmaster** [comment]
Good catch. Yeah, that project is new since the rebase. I've double checked that all sln's open and build.

**Comment by smitpatel** [comment]
:shipit:

File: build/common.props (hunk 3 above)
Comment (bricelam) [reply build/common.props]: *Fool me once...*

**Pull request merged**